diff --git a/cmake/modules/AddLLVM.cmake b/cmake/modules/AddLLVM.cmake
index c8f1083be1f..59b5cfe8340 100644
--- a/cmake/modules/AddLLVM.cmake
+++ b/cmake/modules/AddLLVM.cmake
@@ -87,7 +87,7 @@ function(add_llvm_symbol_exports target_name export_file)
 
     set(CAT "cat")
     set(export_file_nativeslashes ${export_file})
-    if(WIN32 AND NOT CYGWIN)
+    if(${CMAKE_HOST_SYSTEM_NAME} MATCHES "Windows" AND NOT CYGWIN AND NOT MSYS)
       set(CAT "type")
       # Convert ${export_file} to native format (backslashes) for "type"
       # Does not use file(TO_NATIVE_PATH) as it doesn't create a native
diff --git a/cmake/modules/CrossCompile.cmake b/cmake/modules/CrossCompile.cmake
index 400381cda48..189a9a510f0 100644
--- a/cmake/modules/CrossCompile.cmake
+++ b/cmake/modules/CrossCompile.cmake
@@ -28,6 +28,6 @@ if(NOT IS_DIRECTORY ${LLVM_NATIVE_BUILD})
 
   message(STATUS "Configuring native targets...")
   execute_process(COMMAND ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Release
-      -G "${CMAKE_GENERATOR}" -DLLVM_TARGETS_TO_BUILD=${LLVM_TARGETS_TO_BUILD} ${HOST_SYSROOT_FLAGS} ${CMAKE_SOURCE_DIR}
+      ${CROSS_TOOLCHAIN_FLAGS_NATIVE} -G "${CMAKE_GENERATOR}" -DLLVM_TARGETS_TO_BUILD=${LLVM_TARGETS_TO_BUILD} ${HOST_SYSROOT_FLAGS} ${CMAKE_SOURCE_DIR}
     WORKING_DIRECTORY ${LLVM_NATIVE_BUILD} )
 endif(NOT IS_DIRECTORY ${LLVM_NATIVE_BUILD})
