diff --git a/tools/clang/include/clang-c/Platform.h b/tools/clang/include/clang-c/Platform.h
index 5284b533253..0243307fa07 100644
--- a/tools/clang/include/clang-c/Platform.h
+++ b/tools/clang/include/clang-c/Platform.h
@@ -26,7 +26,8 @@ extern "C" {
     #define CINDEX_LINKAGE __declspec(dllimport)
   #endif
 #else
-  #define CINDEX_LINKAGE
+  // otherwise it will be not able to link due all functions are hidden by default
+  #define CINDEX_LINKAGE __attribute__ ((visibility ("default")))
 #endif
 
 #ifdef __GNUC__
