diff --git a/cmake/modules/AddLLVM.cmake b/cmake/modules/AddLLVM.cmake
index 0df6845aaa7..c5cd8b9b219 100644
--- a/cmake/modules/AddLLVM.cmake
+++ b/cmake/modules/AddLLVM.cmake
@@ -1036,7 +1036,7 @@ function(create_subdirectory_options project type)
     if(IS_DIRECTORY "${dir}" AND EXISTS "${dir}/CMakeLists.txt")
       canonicalize_tool_name(${dir} name)
       option(${project}_${type}_${name}_BUILD
-           "Whether to build ${name} as part of ${project}" On)
+           "Whether to build ${name} as part of ${project}" Off)
       mark_as_advanced(${project}_${type}_${name}_BUILD)
     endif()
   endforeach()
